sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventProcessor.java

```diff
@@ -229,14 +236,48 @@ private void receiveEvents(PartitionOwnership partitionOwnership) {
         PartitionProcessor partitionProcessor = this.partitionProcessorFactory
             .createPartitionProcessor(partitionContext, checkpointManager);
         partitionProcessor.initialize().subscribe();
-
+        final AtomicReference<Context> processSpanContext = new AtomicReference<>(Context.NONE);
         consumer.receive().subscribeOn(Schedulers.newElastic("PartitionPump"))
-            .subscribe(eventData -> partitionProcessor.processEvent(eventData).subscribe(unused -> {
-            }, partitionProcessor::processError),
+            .subscribe(eventData -> {
+                startScopedTracingSpan(eventData, processSpanContext);
+                partitionProcessor.processEvent(eventData).subscribe(unused -> {
```

Comment:
I recall this is asyn call `partitionProcessor.processEvent`, right? If so, `endScopedTracingSpan(processSpanContext)` happens before user code returns? Or am I missing something?

Reply:
same as [this](https://github.com/Azure/azure-sdk-for-java/pull/5061#discussion_r317320920)
doOnEach will wait till an item is emitted with success or error signal.